src/photosList/usePhotoList.js

```diff
@@ -6,17 +6,16 @@ export const useFetchedPhotos = (userId) => {
   const [isLoading, setIsLoading] = useState(true);
 
   useEffect(() => {
-    if (userId) {
+    async function getUsersPhotos() {
       setIsLoading(true);
-      api
-        .fetchAlbums(userId)
-        .then((albums) => api.fetchPhotos(albums))
-        .then((photos) => setPhotos(photos))
-        .catch((error) => {
-          console.log("error fetching data", error);
-        })
-        .finally(() => setIsLoading(false));
+      try {
+        const albumResponse = await api.fetchAlbums(userId);
+        const photosResponse = await api.fetchPhotos(albumResponse);
```

Comment:
`api.fetchPhotos` should not care about the albumResponse it should take albumId as an argument and just fetch it. There should be no data sourcing involved in the logic of `api.fetchPhotos`. Also you can rename the function to `api.fetchPhotosByAlbumId`